### ISSUE TITLE
False negatives for consider-using-tuple

### ISSUE THREAD

**Issue by EugeneZelenko** [pylint-dev/pylint#8948]
### Bug description

Looks like `consider-using-tuple` does not analyze `in/not in` operators in conditional statements:

```
name = 'test'
if name in ['value 1', 'value 2']:
    print('found something')
elif name in ['value 3', 'value 4']:
    print('found another thing')
elif name not in ['value 5', 'value 6']:
    print('passed some filter')
```

### Command used

```shell
pylint --load-plugins=pylint.extensions.code_style --enable=all test.py
```

### Pylint output

```shell
************* Module pylint_test
pylint_test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
pylint_test.py:1:0: C0103: Constant name "name" doesn't conform to UPPER_CASE naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 7.14/10 (previous run: 7.14/10, +0.00)
```

### Expected behavior

Warnings should suggest to convert all three lists to tuples.

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.11.4 (main, Jun 26 2023, 16:57:25) [GCC 7.5.0]
```

**Comment by Pierre-Sassoulas** [comment]
I don't remember everything around it, but it's possible that it's volontary because sets are more adequate for membership tests.

**Comment by EugeneZelenko** [comment]
Sure, permanent sets should be better for performance. But is it true for temporary things?

**Comment by Pierre-Sassoulas** [comment]
If you launch pylint with ``pylint test.py --enable=all --enable-all-extentions`` you would get a suggestion to use set: ``2:11: R6201: Consider using set for membership test (use-set-for-membership)``. See https://github.com/pylint-dev/pylint/issues/4776 and attached merge request for full discussion.
